Source/JavaScriptCore/runtime/JSGlobalObject.cpp

```diff
@@ -3268,4 +3282,33 @@ void JSGlobalObject::setWrapperMap(std::unique_ptr<WrapperMap>&& map)
 }
 #endif
 
+void JSGlobalObject::addObjectsForTicket(DeferredWorkTimer::Ticket ticket, JSObject* scriptExecutionOwner, FixedVector<Weak<JSCell>>& dependencies)
+{
+    Locker locker { cellLock() };
+    if (!m_objectsForTicket)
+        m_objectsForTicket = makeUnique<HashMap<DeferredWorkTimer::Ticket, FixedVector<WriteBarrier<JSCell>>>>();
+
+    FixedVector<WriteBarrier<JSCell>> value(dependencies.size() + 1, WriteBarrier<JSCell>());
+    unsigned i = 0;
+    for (auto& dependency : dependencies)
+        value.at(i++).set(vm(), this, dependency.get());
+    value.at(i).set(vm(), this, scriptExecutionOwner);
+
+    m_objectsForTicket->add(ticket, WTFMove(value));
```

Comment:
Let's first add values to the map, and set cells to that to ensure that emitting write-barrier after chaining.
```
auto addResult = m_objectsForTicket->add(ticket, WTFMove(value));
addResult->value.at(I++) = ...
````

Reply:
Thanks, updated.